NewProfile/index.ts

```diff
@@ -0,0 +1,48 @@
+import * as crypto from "crypto";
+import { AzureFunction, Context } from "@azure/functions";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { FiscalCode } from "@pagopa/ts-commons/lib/strings";
+import * as t from "io-ts";
+
+const Failure = t.interface({
+  kind: t.keyof({
+    PERMANENT: null,
+    TRANSIENT: null
+  }),
+  reason: t.string
+});
+type Failure = t.TypeOf<typeof Failure>;
+
+const Success = t.interface({
+  kind: t.literal("SUCCESS")
+});
+type Success = t.TypeOf<typeof Success>;
+
+const permanentDecodeFailure = (errs: t.Errors): Failure =>
+  Failure.encode({
+    kind: "PERMANENT",
+    reason: `Cannot decode input: ${readableReport(errs)}`
+  });
+
+const index: AzureFunction = async (
+  context: Context,
+  inputFiscalCode: unknown
+) =>
+  FiscalCode.decode(inputFiscalCode)
+    .mapLeft(permanentDecodeFailure)
+    .bimap(
+      _ => {
+        context.log.error(`NewProfile|${_.reason}`);
+        return _;
+      },
+      fiscalCode => {
+        // eslint-disable-next-line functional/immutable-data
+        context.bindings.outputFiscalCode = crypto
```

Comment:
instead, we could just define the outbinding as `$return` and just return the payload of the message

Reply:
If we use `$return` we can handle only success cases. We have now a `permanentDecode` error scenario where we don't request a queue retry.